src/main/java/frameworksdrivers/Gateways/SessionDbGateway.java

```diff
@@ -94,11 +95,12 @@ public void saveUser(LoginStudentDbRequestModel requestModel) {
 
             for (Map.Entry<String, Float> course : courses.entrySet()) {
                 String SQL = "INSERT INTO " + DATABASE_NAME_STUDENT +
-                        " (StudentID, CourseID, CourseGrade) VALUES (?, ?, ?)";
+                        " (StudentID, Password, CourseID, CourseGrade) VALUES (?, ?, ?, ?)";
                 PreparedStatement statement = connection.prepareStatement(SQL);
                 statement.setString(1, requestModel.getUsername());
-                statement.setString(2, course.getKey());
-                statement.setFloat(3, course.getValue());
+                statement.setString(2, requestModel.getPassword());
+                statement.setString(3, course.getKey());
+                statement.setFloat(4, course.getValue());
```

Comment:
Now to save a user into the session database. The password of the user is saved.

Reply:
Thanks for listening to my suggestion! I think this will also fix the bug we were having with enrollment as because there wasn't real feasible or efficient way to update all students, with the way it was before students could still take the course and get a grade but it could never be saved so if that course was a prerequisite to a different one the student could never to take the new course because according to the un-updated data base they never took the course